logstash-core/src/main/java/org/logstash/config/ir/CompiledPipeline.java

```diff
@@ -131,9 +131,11 @@ public Dataset buildExecution() {
         outs.forEach(v -> {
             final PluginDefinition def = v.getPluginDefinition();
             final SourceWithMetadata source = v.getSourceWithMetadata();
+            final String sourceFile = v.getSourceFile();
+            final int sourceLine = v.getSourceLine();
             res.put(v.getId(), pluginFactory.buildOutput(
                     RubyUtil.RUBY.newString(def.getName()), RubyUtil.RUBY.newFixnum(source.getLine()),
-                    RubyUtil.RUBY.newFixnum(source.getColumn()), convertArgs(def), convertJavaArgs(def, cve)
+                    RubyUtil.RUBY.newFixnum(source.getColumn()), sourceFile, sourceLine, convertArgs(def), convertJavaArgs(def, cve)
```

Comment:
isn't `source.getLine() == sourceLine` in which case we could be passing down less args?
maybe just Java args (we're changing the signature anyways) and do the Ruby wrapping in `buildOutput`
what do you think?

Reply:
The idea to move string with the source, line and column already formatted into a more meaningful object was an improvement to propose later, maybe after the move of some code from Ruby to Java part (regarding the compilation of pipeline)